### ISSUE TITLE
ADI bug 2: Mismatch between 'sky grid reference and dither reference'

### ISSUE THREAD

**Issue by nswimmer** [MazinLab/MKIDPipeline#85]
This was found when trying to debug issue #84.

While trying to create an output temporal drizzle with `adi_mode: true` and including the `OBJECT` name in the `!MKIDDither: header:` block, drizzler was able to create WCS solutions that it was not able to in #84 (when the object name was not included).

The target of interest (Hip 36152, dithers taken on 22 February 2022) is the first time we have tried to run ADI on a target where the `non-blurring-timestep` is less than the dither `duration` (i.e. the rotation is sufficiently fast that `t_wcs` must be less than `t_bin` in the ADI output). 

When running in ADI mode, drizzler will attempt to drizzle properly, but the error
```
Traceback (most recent call last):
  File "/home/nswimmer/.conda/envs/pipeline/bin/mkidpipe", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/nswimmer/src/mkidpipeline/scripts/mkidpipe", line 113, in <module>
    steps.output.generate(outputs)
  File "/home/nswimmer/src/mkidpipeline/mkidpipeline/steps/output.py", line 79, in generate
    **kwargs)
  File "/home/nswimmer/src/mkidpipeline/mkidpipeline/steps/drizzler.py", line 740, in form
    driz.run(apply_weight=weight)
  File "/home/nswimmer/src/mkidpipeline/mkidpipeline/steps/drizzler.py", line 359, in run
    raise RuntimeError('sky grid ref and dither ref do not match (crval varies between dithers)!')
RuntimeError: sky grid ref and dither ref do not match (crval varies between dithers)!
```
continues to show up.

This appears when trying to create the temporal cube using:
-  `wcs_timestep: non_blurring_minimum` and `timestep=duration of dwell step`
-  `wcs_timestep: non_blurring_minimum` and `timestep<wcs_timestep` and `timestep` is a factor of `duration` (e.g. `duration=15 seconds`, `timestep=1.5 s`)
- `wcs_timestep=timestep < non_blurring_minimum < wcs_timestep` and `timestep` is a factor of `duration` (e.g. `duration=15 seconds`, `timestep=1.5 s`, `wcs_timestep=1.5 s`)

Essentially, every combination of timescales attempted still results in this error. I am at a complete loss with how to generate an ADI cube from this data.

For reference, creating output dithered temporal drizzles with all of the timescales set to the exact same values but with `adi_mode: false` successfully create the output without any problems.

**Comment by baileyji** [comment]
CRVAL is the reference value of the WCS solution. My guess is that this is the same as the #84 and you patching in the object just allowed it to continue, only to find that CRVAL is varying (and shouldn't be). It is made from the various WCS related header keys so I'd check the metadata of file 1 and 2 for changes there.

**Comment by nswimmer** [comment]
So looking at the metadata that creates the WCS for all of the files in the dither, the keys used to make WCS solutions (`'D_IMRDEC', 'D_IMRRA', 'E_PLTSCL', 'E_DPDCX', 'E_DPDCY', 'OBJECT', 'D_IMRPAD'`, per Sarah)don't change from file to file (except for `D_IMRPAD`, which looks like the parallactic angle and I'd expect to change). 

But I think Drizzler is using different values than that? Trying to track that down now.

**Comment by steigersg** [comment]
closed with commit 5e7a4d8
